src/sql/base/browser/ui/panel/panel.component.ts

```diff
@@ -79,23 +87,59 @@ export class PanelComponent extends Disposable {
 	@Input() public actions?: Array<Action>;
 	@ContentChildren(TabComponent) private readonly _tabs!: QueryList<TabComponent>;
 	@ViewChild(ScrollableDirective) private scrollable?: ScrollableDirective;
+	@ViewChild('tabList', { read: ElementRef }) private _tabList!: ElementRef;
 
 	@Output() public onTabChange = new EventEmitter<TabComponent>();
 	@Output() public onTabClose = new EventEmitter<TabComponent>();
 
 	private _activeTab?: TabComponent;
 	private _actionbar?: ActionBar;
 	private _mru: TabComponent[] = [];
+	private _showTabText: boolean = true;
 
 	protected AutoScrollbarVisibility = ScrollbarVisibility.Auto; // used by angular template
 	protected HiddenScrollbarVisibility = ScrollbarVisibility.Hidden; // used by angular template
 	protected NavigationBarLayout = NavigationBarLayout; // used by angular template
 
 	@ViewChild('panelActionbar', { read: ElementRef }) private _actionbarRef!: ElementRef;
-	constructor(@Inject(forwardRef(() => NgZone)) private _zone: NgZone) {
+	constructor(
+		@Inject(forwardRef(() => NgZone)) private _zone: NgZone,
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _cd: ChangeDetectorRef) {
 		super();
 	}
 
+	public get toggleTabLabelButtonClass(): string {
+		return this._showTabText ? 'tab-action collapse' : 'tab-action expand';
+	}
+
+	public get toggleTabLabelButtonLabel(): string {
+		return this._showTabText ? nls.localize('hideTextLabel', "Hide text labels") : nls.localize('showTextLabel', "Show text labels");
+	}
```

Comment:
I'd talk this over with design - personally I'd rather this just be Expand and Collapse. Maybe check what the portal does

Reply:
I copied this from Azure portal :)

Reply:
Are you referring to "Hide/Show the menu for ..."?

![image](https://user-images.githubusercontent.com/28519865/74966517-02956380-53cc-11ea-8c87-f132eacc2510.png)

Run it by design, if they're fine with this text then that's fine. I'm just not a huge fan of the whole "text label" thing. Just seems overly verbose.

Reply:
![Feb-20-2020 10-55-48](https://user-images.githubusercontent.com/13777222/74968388-9caadb00-53cf-11ea-8230-f30910a9b1d0.gif)

Reply:
Ah cool, should have looked at the actual dashboard! Alright, I'd still run this by PM/Design to verify but in the end I'm fine if this stays like this.